pkg/kudoctl/kudoinit/setup/setup.go

```diff
@@ -33,14 +36,22 @@ func NewInstaller(options kudoinit.Options, crdOnly bool) *Installer {
 		}
 	}
 
+	// This is a bit cumbersome - we need to access some funcs from these two
+	// steps for the upgrade process, that's why they are initialized here.
+	// This should be cleaned up
```

Comment:
could you elaborate (in code comments) what is meant by cleaned up?

Reply:
I've removed the "Cleaned up" part and changed the comment a bit. 

I think the whole `setup.go` with the `[]kudoinit.Step` slice should probably refactored. It was a good idea in the beginning, but doesn't work that well with upgrades when we need to access specific elements from the slice.